SayTheirNames/Source/Controller/Donations/DonationsMoreDetails/DonationsMoreDetailsController.swift

```diff
@@ -98,21 +137,36 @@ final class DonationsMoreDetailsController: UIViewController {
 
         view.accessibilityIdentifier = AccessibilityIdentifers.view
         donationButtonContainerView.accessibilityIdentifier = AccessibilityIdentifers.donationButtonContainerView
-        
-        // Fetch Donation Details
-        self.network.fetchDonationDetails(with: donation.identifier) { [weak self] in
-            switch $0 {
-            case .success(let page):
-                self?.configure(with: page.donation)
-
-            case .failure(let error):
-                Log.print(error)
-             }
+             
+        switch data {
+            
+        case .petition(let petition):
+            self.network.fetchPetitionDetails(with: petition.identifier) { [weak self] (result) in
+                switch result {
+                case .success(let page):
+                    self?.configure(with: Data.petition(page.petition))
+                    
+                case .failure(let error):
+                    Log.print(error)
```

Comment:
We should add error states at some point 🙈

Reply:
would be nice yeah 🙈  wasn't the top priority for anyone so far 😬